### PR TITLE
BC-724 - Listing elements and search

### PR DESCRIPTION
**Pull request by murat-merdoglu-hpi**
# Short Description
- Implemented all elements' list page
- Added search feature to list page
- Added search feature to the dashboard
- Created a new iterator elements' component

## Links to Ticket and related Pull-Requests
https://ticketsystem.hpi-schul-cloud.org/browse/BC-724

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/pages/rooms-overview.vue [hunk 1]
```diff
@@ -105,30 +134,38 @@ export default {
 			return this.$t("common.labels.greeting", { name: this.$user.firstName });
 		},
 	},
-	async created() {
+	async mounted() {
 		await RoomsModule.fetch(); // TODO: this method will receive a string parameter (Eg, mobile | tablet | desktop)
```
Comment (pkleybolte) [inline src/pages/rooms-overview.vue:138]: TODO is not handled in this ticket? Is there a follow up ticket you could mention here?

File: src/pages/rooms-overview.vue [hunk 2]
```diff
@@ -181,7 +221,8 @@ export default {
 					this.draggedElementName == "groupItem") &&
 				toElementName == "vRoomAvatar"
 			) {
-				this.savePosition();
+				await this.savePosition();
+				// TODO: default naming
```
Comment (pkleybolte) [inline src/pages/rooms-overview.vue:225]: TODO is not handled in this ticket? Is there a follow up ticket you could mention here?

File: src/pages/rooms-overview.vue [hunk 3]
```diff
@@ -228,6 +269,24 @@ export default {
 			this.roomsData = RoomsModule.getRoomsData;
 			this.groupDialog.groupData = {};
 		},
+		searchItems(filterText) {
```
Comment (pkleybolte) [inline src/pages/rooms-overview.vue:272]: Nice to have:
As computed property.

File: src/serverMiddleware/routes.js [hunk 4]
```diff
@@ -39,6 +39,7 @@ const routes = [
 	`^/news/${mongoId}/edit`,
 	`^/nuxtversion/?$`,
 	`^/rooms-overview/?$`,
+	`^/rooms-list/?$`,
```
Comment (pkleybolte) [inline src/serverMiddleware/routes.js:42]: Is the task to open this route handled?

File: src/store/rooms.ts [hunk 5]
```diff
@@ -11,7 +11,40 @@ import {
 	DashboardApiFactory,
 	DashboardApiInterface,
 } from "../serverApi/v3/api";
-import { DroppedObject, RoomsData } from "./types/rooms";
```
Comment (pkleybolte) [inline src/store/rooms.ts:14]: Types are externalized to the /types folder in another ticket.
Please do it for these and the following types as well.

File: src/store/rooms.ts [hunk 6]
```diff
@@ -152,6 +195,19 @@ export class Rooms extends VuexModule {
 			this.setLoading(false);
 		}
 	}
+
+	@Action
+	async fetchAllElements(): Promise<void> {
+		this.setLoading(true);
+		try {
+			const { data } = await $axios.$get("/v3/courses/");
```
Comment (pkleybolte) [inline src/store/rooms.ts:203]: If it is api/v3 there should be a generated client available that we can use here.

**Review by pkleybolte — CHANGES_REQUESTED** [review]

File: src/pages/rooms-overview.vue (hunk 1 above)
Comment (murat-merdoglu-hpi) [reply src/pages/rooms-overview.vue:138]: The server has not supported a device-based dashboard yet, and there is no open ticket by now.

File: src/serverMiddleware/routes.js (hunk 4 above)
Comment (murat-merdoglu-hpi) [reply src/serverMiddleware/routes.js:42]: https://github.com/hpi-schul-cloud/dof_app_deploy/pull/62

File: src/store/rooms.ts (hunk 5 above)
Comment (murat-merdoglu-hpi) [reply src/store/rooms.ts:14]: done

File: src/pages/rooms-overview.vue (hunk 3 above)
Comment (murat-merdoglu-hpi) [reply src/pages/rooms-overview.vue]: done

File: src/pages/rooms-list.vue [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+		title() {
+			return this.$t("common.labels.greeting", { name: this.$user.firstName });
+		},
+	},
+	async mounted() {
+		await RoomsModule.fetchAllElements();
+		this.$data.allElements = RoomsModule.getAllElements;
+	},
+	methods: {
+		searchItems(filterText: string): void {
```
[the remark this answers is not in the document]
Comment (murat-merdoglu-hpi) [reply src/pages/rooms-list.vue]: done

**Review by pkleybolte — APPROVED** [review]

**Pull request merged**